common/app/views/fragments/preRenderJavaScript.scala.html

```diff
@@ -0,0 +1,58 @@
+@()
+
+@* This script must come after the pre-flight shims *@
+
+<script>
+(function applyAdFree(){
+    /**
+     * Determine if user has 'adfree' view, and apply appropriate body classes before rendering.
+     */
+
+    var ADFREE_COOKIE_KEY = 'gu_adfree_user',
+        USER_SIGNIN_COOKIE = 'GU_U',
+        commercial = guardian.config.commercial;
+
+    if (showAdfreeView()) {
+        addClass(document.documentElement, 'adfree');
+        commercial.showingAdfree = true;
+    } else {
+        commercial.showingAdfree = false;
+    }
+
+    function showAdfreeView() {
+        if (featureDisabled() || userLoggedOut()) {
+            return false;
+        } else {
+            var adfreeUser = readCookie(ADFREE_COOKIE_KEY);
+            // If the cookie is ambiguous or null, err on the side of caution and hold off showing ads
+            return adfreeUser !== 'false';
+        }
+    }
+
+    function featureDisabled() {
+        return guardian.config.switches.advertOptOut === false;
```

Comment:
No, no, I think that makes complete sense. In fact, we could wrap the whole IIFE itself with a switch in the featureDetection file (and bring in the IIFE as a fragment). That way I (believe) the function declarations themselves won't even be evaluated (I think the engine only does lookahead when it enters the function execution context, but I might be wrong there)